app/models/user.rb

```diff
@@ -94,6 +94,10 @@ def self.leader_senior_voters
     (team_leaders + senior_delegates).uniq.compact
   end
 
+  def self.regional_voters
+    regional_delegates = RolesMetadataDelegateRegions.regional_delegate.where(user_roles: { end_date: nil }).includes(:user).map(&:user).uniq.compact
```

Comment:
It seems that you did not address my previous comment about re-using the scope with `.joins(:user_roles).merge(UserRole.active)`. Did you try it, and did it work?